prereise/gather/griddata/hifld/data_process/transmission.py

```diff
@@ -0,0 +1,171 @@
+from powersimdata.utility.distance import haversine
+
+from prereise.gather.griddata.hifld.data_access import load
+
+
+def check_for_location_conflicts(substations):
+    """Check for multiple substations with identical lat/lon.
+
+    :param pandas.DataFrame substations: data frame of substations.
+    :raises ValueError: if multiple substations with identical lat/lon.
+    """
+    num_substations = len(substations)
+    num_lat_lon_groups = len(substations.groupby(["LATITUDE", "LONGITUDE"]))
+    if num_lat_lon_groups != num_substations:
+        num_collisions = num_substations - num_lat_lon_groups
+        raise ValueError(
+            f"There are {num_collisions} substations with duplicate lat/lon values"
+        )
+
+
+def get_substations_with_nonzero_lines(substations):
```

Comment:
I was comparing the [clean](https://github.com/Mingze-Li128/PreREISE/blob/31693d4121a7fa4e856a3567124e8f34b0caab8c/prereise/gather/hiflddata/data_trans.py#L99) function with this and found they didn't filer out not "IN SERVICE" substations. I remember we discussed this before, just wanna get an impression how many substations we filtered out?

Reply:
Filtering to only `IN SERVICE` yields 72,091 substations, out of a starting total of 75,328.

Reply:
That's more than 3k substations. In this way, I presume our output csvs will be much different from theirs.

Reply:
This filtering originally went in via @rouille's #196. I agree that we will end up with a different set of substations. However, based on previous conversations, there are also other changes we anticipate making, so I expect that our code will produce different output CSVs for the bus, branch, and plant tables as well. I guess there are two questions:
1. Do we want to be including substations with statuses like `INACTIVE` (14), `NOT AVAILABLE` (3178), `PROPOSED` (5), and `UNDER CONST` (40)?
2. Should our PRs into the `hifld` branch be based on what we currently think are the most reasonable assumptions, or based on a faithful reproduction of the old branch (with potential improvements after all existing functionality is integrated)?

Reply:
1. No
2. Yes, it should definitely based on what we currently think are the most reasonable assumptions.
I put the note down here just in case if we would like to compare our csvs with theirs at some point in the future, here is one position where we start deviating from their track..

Reply:
What is the meaning of `NOT AVAILABLE`? Would it be useful to toggle some filters so we can test some combinations?

Reply:
I'm not sure the meaning of the `NOT AVAILABLE` substations. Since this filtering comes from the `load` module, I think modifying it is out of scope of this PR.

Reply:
Sure, I was just wondering if that would be of use. But `NOT AVAILABLE` is a small percentage so maybe that is not that useful then.
Regarding your two questions I agree with @BainanXia